### ISSUE TITLE
Random failure on CI

### ISSUE THREAD

**Issue by tagliala** [igorkasyanchuk/active_storage_validations#266]
Ref: https://github.com/igorkasyanchuk/active_storage_validations/actions/runs/11288843453/job/31397452279?pr=265#step:7:141

Got it on both 7.2/vips and 7.0 mini_magick

```
Error:
ActiveStorageValidations::Test#test_dimensions_with_attached_StringIO:
Errno::ENOENT: No such file or directory @ rb_file_s_size - /tmp/image_1920x108020241011-2769-7xqr6w.png
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image/info.rb:51:in `size'
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image/info.rb:51:in `block in cheap_info'
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image/info.rb:40:in `fetch'
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image/info.rb:40:in `cheap_info'
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image/info.rb:17:in `[]'
    vendor/bundle/ruby/3.3.0/gems/mini_magick-5.0.1/lib/mini_magick/image.rb:123:in `block in attribute'
    lib/active_storage_validations/metadata.rb:32:in `block in metadata'
    lib/active_storage_validations/metadata.rb:98:in `read_image'
    lib/active_storage_validations/metadata.rb:28:in `metadata'
    lib/active_storage_validations/dimension_validator.rb:66:in `block in validate_each'
    lib/active_storage_validations/dimension_validator.rb:65:in `each'
    lib/active_storage_validations/dimension_validator.rb:65:in `validate_each'
    vendor/bundle/ruby/3.3.0/gems/activemodel-7.0.8.4/lib/active_model/validator.rb:153:in `block in validate'
    vendor/bundle/ruby/3.3.0/gems/activemodel-7.0.8.4/lib/active_model/validator.rb:[149](https://github.com/igorkasyanchuk/active_storage_validations/actions/runs/11288843453/job/31397452279?pr=265#step:7:150):in `each'
```

**Comment by Mth0158** [comment]
Hi @tagliala
I have no idea yet why it's behaving this way, it could be a regression due to the latest merged PR (https://github.com/igorkasyanchuk/active_storage_validations/pull/256). I'll dedicate some time this week to look into it, thks for the issue!

**Comment by tagliala** [comment]
Hi, thanks.

I think it may depend by `bundle update` for some reasons. I'm checking too

**Comment by Mth0158** [comment]
@tagliala FYI it was happening due to our tempfile management system inside the `metadata.rb` file.
It will be fixed by the following refacto PR #299
